src/base/UnorderedNonce.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.24;
+
+/// @title Unordered Nonce
+/// @notice Contract state and methods for using unordered nonces in signatures
+contract UnorderedNonce {
+    error NonceAlreadyUsed();
+
+    mapping(address owner => mapping(uint256 word => uint256 bitmap)) public nonces;
+
+    /// @notice Consume a nonce, reverting if its already been used
+    /// @param owner address, the owner/signer of the nonce
+    /// @param nonce uint256, the nonce to consume
+    function _useUnorderedNonce(address owner, uint256 nonce) internal {
+        (uint256 wordPos, uint256 bitPos) = bitmapPositions(nonce);
+        uint256 bit = 1 << bitPos;
```

Comment:
technically we have a gas optimised version of this in core if it interests you
https://github.com/Uniswap/v4-core/blob/799dd2cb980319a8d3b827b6a7aa59a606634553/src/libraries/TickBitmap.sol#L48-L52

Reply:
took at crack at this, so please review closely ❤️

Reply:
it looks great!!! 👯